### ISSUE TITLE
Ability to change tokens lifespans on the fly

### ISSUE THREAD

**Issue by randaalex** [ory/fosite#473]
Currently, lifespans have static behavior (getter just read value from config)
But I would like to have different lifespans per client (for example). 

So it would be more comfortable to have something like `handler.GetAccessTokenLifespan(ctx, request)` instead of  `handler.AccessTokenLifespan`

**Comment by mitar** [comment]
I think you can do that. You just have to manually set the claim on the token. Config ones is used only if claim for expiration does not yet exist.

**Comment by randaalex** [comment]
@mitar yes, it will work for access token. but my proposal about all tokens (refresh, auth, id)

**Comment by mitar** [comment]
I didn't get that from the original issue description. But I think you can do the same for other tokens as well? For ID for sure, auth is not a token but a code. For refresh I am not sure.

**Comment by aeneasr** [comment]
What's your use case? Access tokens have typically static expiry times because you get refresh tokens, which you can use to refresh access tokens.

**Comment by randaalex** [comment]
@mitar sorry for ambiguity
@aeneasr I need ability to have different refresh-tokens TTLs per client

**Comment by mitar** [comment]
See also: https://github.com/ory/fosite/issues/211

I do think we need better documentation here how to do few common things.

**Comment by aeneasr** [comment]
I think client-specific TTL is ok. What we need to figure out is how to prevent clients from defining their own refresh token TTL, which should be only allowed for server administrators because it is considered server configuration.

**Comment by mitar** [comment]
> What we need to figure out is how to prevent clients from defining their own refresh token TTL,

**Comment by aeneasr** [comment]
> Even if it is less than what server/admin configured?

In that case I think it would be ok!
